flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DefaultSchedulerFactory.java

```diff
@@ -66,14 +77,30 @@ public SchedulerNG createInstance(
 			jobMasterConfiguration,
 			slotProvider,
 			futureExecutor,
+			new ScheduledExecutorServiceAdapter(futureExecutor),
 			userCodeLoader,
 			checkpointRecoveryFactory,
 			rpcTimeout,
 			blobWriter,
 			jobManagerJobMetricGroup,
 			slotRequestTimeout,
 			shuffleMaster,
-			partitionTracker);
+			partitionTracker,
+			schedulingStrategyFactory,
+			new RestartPipelinedRegionStrategy.Factory(),
```

Comment:
I think we should first get the scheduler to a working state.

Reply:
We can do this, but I was wondering who would actually want to use these restart strategies? Shouldn't the pipelined region restart strategy be more efficient in most cases?

Reply:
@tillrohrmann an individual restart strategy is needed if we want to only restart the failed task for recovery. Ot maybe just provide a pluggable way so that developer can do it out of flink core. Here's the [ML](https://lists.apache.org/thread.html/237de1061283a18dd813043a84d5d1debb15d1220368fa2f5b801025@%3Cdev.flink.apache.org%3E) for individual failover discussion.

Reply:
And a FastRestartPipelinedRegionStrategy is introduced in FLINK-13056 as an option for faster failover processing.

Reply:
All right, I see. Let's do this as a follow up issue.

Reply:
I opened a ticket FLINK-14131 for this. We can do it after this PR is done.

Add we need to consider whether to respect the existing "jobmanager.execution.failover-strategy" config then.